datatypes/src/dataset.rs

```diff
@@ -64,3 +61,297 @@ impl From<ExternalDataId> for DataId {
         DataId::External(value)
     }
 }
+
+pub const NAME_DELIMITER: char = ':';
+pub const SYSTEM_NAMESPACE: &str = "_";
+pub const SYSTEM_PROVIDER: &str = "_";
+
+/// The user-facing identifier for loadable data.
+/// It can be resolved into a [`DataId`].
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct NamedData {
+    pub namespace: Option<String>,
+    pub provider: Option<String>,
+    pub name: String,
+}
+
+impl NamedData {
+    /// Canonicalize a name that reflects the system namespace and provider.
+    fn canonicalize<S: Into<String> + PartialEq<&'static str>>(
+        name: S,
+        system_name: &'static str,
+    ) -> Option<String> {
+        if name == system_name {
+            None
+        } else {
+            Some(name.into())
+        }
+    }
+
+    pub fn with_global_name<S: Into<String>>(name: S) -> Self {
+        Self {
+            namespace: None,
+            provider: None,
+            name: name.into(),
+        }
+    }
+
+    pub fn with_namespaced_name<S1: Into<String> + PartialEq<&'static str>, S2: Into<String>>(
+        namespace: S1,
+        name: S2,
+    ) -> Self {
+        Self {
+            namespace: Self::canonicalize(namespace, SYSTEM_NAMESPACE),
+            provider: None,
+            name: name.into(),
+        }
+    }
+
+    pub fn with_global_provider<S1: Into<String> + PartialEq<&'static str>, S2: Into<String>>(
```

Comment:
can you add a comment what a global provider is?

Reply:
[6e97ae8](https://github.com/geo-engine/geoengine/pull/782/commits/6e97ae81f0f45f0fd46f095e57ae0b6238f9e15d)